AttitudeManager/Src/MovingAverage.cpp

```diff
@@ -0,0 +1,41 @@
+#include "MovingAverage.hpp"
+#include <string.h>
+
+template<typename T>
+Window<T>::Window() {
+  memset(window, 0, MAX_WINDOW_SIZE);
+  size = 0;
+  pos = 0;
+}
+
+template<typename T>
+void Window<T>::update(T newValue) {
+  window[pos] = newValue;
+  pos++;
+
+  // If the window is not full yet, increase size.
+  if (size < MAX_WINDOW_SIZE) {
+    size++;
+  }
+
+  // Reset pos once in a while to ensure it doesn't get too big.
+  if (pos == MAX_POS_LIMIT) {
+    pos = 0;
+  }
+}
+
+template<typename T>
+MovingAverageError_t Window<T>::getAverage(uint8_t n, T* result) {
+  if (n <= 0 || n > size) {
+    return FAILURE;
+  }
+
+  int sum = 0;
+  for (int i = pos; i > pos - n; i--) {
+    uint8_t realIdx = pos % MAX_WINDOW_SIZE;
+    sum += window[realIdx];
+  }
+
+  *result = static_cast<T>(sum) / n;
```

Comment:
I think the idea for the static cast is good - the data that IMU's deal with is almost always floating point decimal (i.e. 0.9g, 1.1g, 1.7137g, etc), so it might be worth forcing a float return.

Something about safety-critical applications and never "guessing" your return type. Search for "cast" in this doc: https://embeddedgurus.com/barr-code/category/coding-standards/page/2/ as a starting point